File: dailyprogrammer/2/easy/main.js
'use strict';

const readlineSync = require('readline-sync');

let conversions = {
  f: {
    c: (currentTemp) => {
      return (currentTemp - 32) * 5 / 9;
    },
    k: (currentTemp) => {
      return conversions['f']['c'](current_temp) + 273.15;
    }
  },
  c: {
    f: (currentTemp) => {
      return (currentTemp * 9 / 5) + 32;
    },
    k: (currentTemp) => {
      return currentTemp + 273.15;
    }
  },
  k: {
    c: (currentTemp) => {
      return currentTemp - 273.15;
    },
    f: (currentTemp) => {
      return conversions['c']['f'](current_temp - 273.15);
    }
  }
};

function tip(amount, percentage) {
  return total + (total * percentage / 100);
}

let action = readlineSync.question('Tip or temperature?\n');

if (action.lower() === 'tip') {
  let total = parseFloat(readlineSync.question('What is the total?\n'));
  let rate = parseFloat(readlineSync.question('What is the tipping rate?\n'));
  console.log(tip(total, rate));
} else {
  let currentTemp = parseFloat(readlinesync.question('What is the current temperature\n'));
  let fromUnit = readlinesync.question('What is the current unit?\n');
  let toUnit = readlinesync.question('What unit should it be convered to?\n');
  console.log(conversions[fromUnit][toUnit](currentTemp));
}

File: dailyprogrammer/2/easy/main.py
def tip_calculator(total, percent_int):
    return int(total) + ((int(total) * int(percent_int)) / 100)


def temperature_conversion(current_temp, current_unit, to_unit):
    conversions = {
    	'f': {
    		'c': f_to_c,
    		'k': f_to_k
    	},
    	'c': {
    		'f': c_to_f,
    		'k': c_to_k
    	},
    	'k': {
    		'c': k_to_c,
    		'f': k_to_f
    	}
    }
    
    return conversions[current_unit][to_unit](current_temp)


def c_to_k(current_temp):
    return current_temp + 273.15


def k_to_c(current_temp):
    return current_temp - 273.15


def f_to_c(current_temp):
    return (current_temp - 32) * 5 / 9


def c_to_f(current_temp):
    return (current_temp * 9 / 5) + 32


def f_to_k(current_temp):
    return f_to_c(current_temp) + 273.15


def k_to_f(current_temp):
    return c_to_f(current_temp - 273.15)


def main():
    choice = input('Tip or temperature?\n')
    if choice.lower() == 'tip':
        total = input('What is the total?\n')
        percent_int = input('What is the tipping rate?\n')
        print(tip_calculator(total, percent_int))
    elif choice.lower() == 'temperature':
        current_temp = float(input('What is the current temperature?\n'))
        current_unit = input('What is the current unit?\n')
        to_unit = input('What unit should it be changed to? \n')
        print(temperature_conversion(current_temp, current_unit, to_unit))

if __name__ == '__main__':
    main()
